timemachine/fe/atom_mapping.py

```diff
@@ -40,12 +51,22 @@ def mcs(a, b, threshold: float = 2.0, timeout: int = 5, smarts: Optional[str] =
     if conformer_aware:
         params.AtomCompareParameters.MaxDistance = threshold
     params.AtomTyper = rdFMCS.AtomCompare.CompareAny
-
     # globals
     params.Timeout = timeout
     if smarts is not None:
         params.InitialSeed = smarts
 
+    def strip_hydrogens(mol):
+        """Strip hydrogens with deepcopy to be extra safe"""
+        return Chem.RemoveHs(deepcopy(mol))
+
+    if not match_hydrogens:
+        # Setting CompareAnyHeavyAtom doesn't handle this correctly, strip hydrogens explicitly
```

Comment:
If possible, are there any details to share on how `CompareAnyHeavyAtom` fails to restrict the MCS to heavy atoms only?

Reply:
(Sorry, meant to delete this comment before posting review-- resolving.)